### DISCUSSION TITLE
Navigation in search mode

### DISCUSSION THREAD

**Discussion by max-baz** [sayanarijit/xplr#247] [Q&A]
In xplr repo folder, run `$ touch s`, then open xplr.
Say you want to navigate to the file called `s`, you press `/s` and you still see a lot of other files. How do you navigate to `s`?
Arrows work of course, I intutively tried to press `$` (like a regex, to say that `s` is the last letter in the name), I tried to press `PageDown` and `End` and `ArrowUp`, but they do nothing.

Am I missing some trick? 🙂 Not sure about regex support, maybe an overkill for search, but at least `PageDown` and `End` we could add? With `ArrowUp` also not sure, sometimes I love when cursor wraps, sometimes I hate it 🙂

**Comment by sayanarijit** [comment]
There are messages like starts with and ends with. You can probably match some characters and make them behave like regex. Let me know if you need help with the implementation.

**Reply by max-baz** [reply]
OK, although regex probably is the least intuitive way, what do you think of PageDown and/or End buttons, and maybe wrapping with ArrowUp to go the the end?

**Reply by sayanarijit** [reply]
I don't have those keys in my laptop so can't say I ever used it. What do they do?

**Reply by sayanarijit** [reply]
Ok I actually have them. But as combination keys.

**Reply by sayanarijit** [reply]
Seems useful. Let's have them.

**Reply by sayanarijit** [reply]
i.e. PageUp and PageDown.

**Reply by max-baz** [reply]
PageDown would go one page (or often also half a page) down, like `Ctrl+d`/`Ctrl+u` in vim, End would go to the end like `G` does (you can't press G to go the end in search mode, so that's why I thought of End), wrapping is available in nnn so you can try there (open nnn and press arrow Up, it will go to the last file)

**Reply by sayanarijit** [reply]
It seems useful in search mode, but conflicts with the default mode behavior. The search mode works more like fzf which doesn't have this feature, probably for a good reason.

**Reply by sayanarijit** [reply]
I think I can make FocussPrevious and FocusNext specifically work like that, i.e. after max index, continue with 0. But both in default and search mode. What do you think?

**Reply by max-baz** [reply]
Yeah makes sense to have consistency everywhere, lets do it!

**Reply by sayanarijit** [reply]
https://github.com/sayanarijit/xplr/issues/253
